### ISSUE TITLE
bcftools make error

### ISSUE THREAD

**Issue by starsyi** [samtools/bcftools#422]
**The following is my make process:**
```
[usr@login bcftools]$ make
cd ../htslib && make config.h
make[1]: Entering directory `/public/home/usr/bin/htslib'
echo '/* Empty config.h generated by Makefile */' > config.h
make[1]: Leaving directory `/public/home/usr/bin/htslib'
cd ../htslib && make lib-static
make[1]: Entering directory `/public/home/usr/bin/htslib'
gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c
gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c
gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c
gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c
gcc -g -Wall -O2 -I.  -c -o errmod.o errmod.c
gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c
gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c
gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c
echo '#define HTS_VERSION "1.3.1-14-gf2399ee"' > version.h
gcc -g -Wall -O2 -I.  -c -o hts.o hts.c
gcc -g -Wall -O2 -I.  -c -o md5.o md5.c
gcc -g -Wall -O2 -I.  -c -o probaln.o probaln.c
gcc -g -Wall -O2 -I.  -c -o realn.o realn.c
gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c
gcc -g -Wall -O2 -I.  -c -o sam.o sam.c
gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c
gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c
gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c
gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c
gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c
gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c
gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c
gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c
gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c
gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c
gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c
gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c
gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c
gcc -g -Wall -O2 -I.  -c -o cram/thread_pool.o cram/thread_pool.c
gcc -g -Wall -O2 -I.  -c -o cram/vlen.o cram/vlen.c
gcc -g -Wall -O2 -I.  -c -o cram/zfio.o cram/zfio.c
ar -rc libhts.a kfunc.o knetfile.o kstring.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o md5.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o cram/thread_pool.o cram/vlen.o cram/zfio.o
ranlib libhts.a
make[1]: Leaving directory `/public/home/usr/bin/htslib'
echo '#define BCFTOOLS_VERSION "1.3.1-4-ge3ef8da"' > version.h
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o main.o main.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfindex.o vcfindex.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o tabix.o tabix.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfstats.o vcfstats.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfisec.o vcfisec.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfmerge.o vcfmerge.c
vcfmerge.c: In function ‘info_rules_init’:
vcfmerge.c:277:9: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
         while ( *ss ) ss++; ss++;
         ^~~~~
vcfmerge.c:277:29: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘while’
         while ( *ss ) ss++; ss++;
                             ^~
vcfmerge.c:303:9: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
         while ( *ss ) ss++; ss++; n++;
         ^~~~~
vcfmerge.c:303:29: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘while’
         while ( *ss ) ss++; ss++; n++;
                             ^~
vcfmerge.c: In function ‘debug_maux’:
vcfmerge.c:1661:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
     for (j=0; j<maux->nals; j++) fprintf(stderr,"%s   %dx %s", j==0?"":",",maux->cnt[j], maux->als[j]); fprintf(stderr,"\n");
     ^~~
vcfmerge.c:1661:105: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘for’
     for (j=0; j<maux->nals; j++) fprintf(stderr,"%s   %dx %s", j==0?"":",",maux->cnt[j], maux->als[j]); fprintf(stderr,"\n");
                                                                                                         ^~~~~~~
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfquery.o vcfquery.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcffilter.o vcffilter.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o filter.o filter.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfsom.o vcfsom.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfnorm.o vcfnorm.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfgtcheck.o vcfgtcheck.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfview.o vcfview.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfannotate.o vcfannotate.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfroh.o vcfroh.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfconcat.o vcfconcat.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfcall.o vcfcall.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o mcall.o mcall.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcmp.o vcmp.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o gvcf.o gvcf.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o reheader.o reheader.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o convert.o convert.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfconvert.o vcfconvert.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o tsv2vcf.o tsv2vcf.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfcnv.o vcfcnv.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o HMM.o HMM.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o vcfplugin.o vcfplugin.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o consensus.o consensus.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o ploidy.o ploidy.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o version.o version.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o ccall.o ccall.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o em.o em.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o prob1.o prob1.c
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o kmin.o kmin.c
gcc -rdynamic  -o bcftools main.o vcfindex.o tabix.o vcfstats.o vcfisec.o vcfmerge.o vcfquery.o vcffilter.o filter.o vcfsom.o vcfnorm.o vcfgtcheck.o vcfview.o vcfannotate.o vcfroh.o vcfconcat.o vcfcall.o mcall.o vcmp.o gvcf.o reheader.o convert.o vcfconvert.o tsv2vcf.o vcfcnv.o HMM.o vcfplugin.o consensus.o ploidy.o version.o ccall.o em.o prob1.o kmin.o  ../htslib/libhts.a -lpthread -lz -lm -ldl  
gcc -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"  -c -o test/test-rbuf.o test/test-rbuf.c
test/test-rbuf.c: In function ‘debug_print’:
test/test-rbuf.c:32:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
     for (i=-1; rbuf_next(rbuf, &i); ) printf(" %2d", i); printf("\n");
     ^~~
test/test-rbuf.c:32:58: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘for’
     for (i=-1; rbuf_next(rbuf, &i); ) printf(" %2d", i); printf("\n");
                                                          ^~~~~~
test/test-rbuf.c:33:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
     for (i=-1; rbuf_next(rbuf, &i); ) printf(" %2d", dat[i]); printf("\n");
     ^~~
test/test-rbuf.c:33:63: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘for’
     for (i=-1; rbuf_next(rbuf, &i); ) printf(" %2d", dat[i]); printf("\n");
                                                               ^~~~~~
gcc  -o test/test-rbuf test/test-rbuf.o -lm 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/color-chrs.so HMM.c version.c plugins/color-chrs.c 
plugins/color-chrs.c: In function ‘flush_viterbi’:
plugins/color-chrs.c:468:9: warning: ‘s2’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         fprintf(args->fp,"# SG, shared segment\t[2]Chromosome\t[3]Start\t[4]End\t[5]%s:1\t[6]%s:2\n",s2,s2);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
plugins/color-chrs.c:531:5: warning: ‘s1’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     fprintf(args->fp,"SW\t%s\t%s\t%d\t%d\t%f\n", s1,bcf_hdr_id2name(args->hdr,args->prev_rid),args->nhet_mother,nswitch_mother,mrate);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/counts.so version.c plugins/counts.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/dosage.so version.c plugins/dosage.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/fill-AN-AC.so version.c plugins/fill-AN-AC.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/fill-tags.so version.c plugins/fill-tags.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/fixploidy.so ploidy.c version.c plugins/fixploidy.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/frameshifts.so version.c plugins/frameshifts.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/GTisec.so version.c plugins/GTisec.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/impute-info.so version.c plugins/impute-info.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/mendelian.so version.c plugins/mendelian.c 
plugins/mendelian.c: In function ‘init’:
plugins/mendelian.c:149:13: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
             while ( *se && *se!=',' ) se++; if ( *se!=',' ) error("Could not parse %s: %s\n",trio_file, ss);
             ^~~~~
plugins/mendelian.c:149:45: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘while’
             while ( *se && *se!=',' ) se++; if ( *se!=',' ) error("Could not parse %s: %s\n",trio_file, ss);
                                             ^~
plugins/mendelian.c:154:13: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
             while ( *se && *se!=',' ) se++; if ( *se!=',' ) error("Could not parse %s\n",trio_file);
             ^~~~~
plugins/mendelian.c:154:45: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘while’
             while ( *se && *se!=',' ) se++; if ( *se!=',' ) error("Could not parse %s\n",trio_file);
                                             ^~
plugins/mendelian.c:159:13: warning: this ‘while’ clause does not guard... [-Wmisleading-indentation]
             while ( *se ) se++; if ( ss==se ) error("Could not parse %s\n",trio_file);
             ^~~~~
plugins/mendelian.c:159:33: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘while’
             while ( *se ) se++; if ( ss==se ) error("Could not parse %s\n",trio_file);
                                 ^~
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/missing2ref.so version.c plugins/missing2ref.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/setGT.so version.c plugins/setGT.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/tag2tag.so version.c plugins/tag2tag.c 
gcc -fPIC -shared -g -Wall -Wc++-compat -O2 -I. -I../htslib -DPLUGINPATH=\"/usr/local/libexec/bcftools\"   -o plugins/vcf2sex.so ploidy.c version.c plugins/vcf2sex.c 
plugins/vcf2sex.c: In function ‘run’:
plugins/vcf2sex.c:415:17: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
                 if (*tmp) error("Unexpected argument to --nsites: %s\n", optarg); break;
                 ^~
plugins/vcf2sex.c:415:83: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
                 if (*tmp) error("Unexpected argument to --nsites: %s\n", optarg); break;
                                                                                   ^~~~~
```
**I want to know, if it has make succeed? and if it have some effects**

**Comment by pd3** [comment]
These are all warnings, not errors, so you should be fine.

**Comment by mp15** [comment]
This is a gcc 6 new warning but we should probably clean these up at some point.

**Comment by mcshane** [comment]
Closed by ee8148d31f97c39c77fd7d5b4a4e1d5ca8408031
